lib/buffered_logger/log_device_proxy.rb

```diff
@@ -25,12 +25,12 @@ def start
     end
 
     def started?
-      @buffers.key?(key)
+      @buffers.clone.key?(key)
```

Comment:
I don't think this change is necessary; the change to `sweep` seems to be enough.